server/api/jurisdictions.py

```diff
@@ -804,9 +814,110 @@ def get_batch_comparison_audit_discrepancies_by_jurisdiction(
             jurisdiction_name, batch_name = batch_key
             jurisdiction_id = jurisdiction_name_to_id[jurisdiction_name]
             discrepancies_by_jurisdiction[jurisdiction_id][batch_name][contest.id] = {
-                "reportedVotes": reported_contest_result,
-                "auditedVotes": audited_contest_result,
-                "discrepancies": vote_deltas,
+                "reportedVotes": normalize_vals_for_discrepancies(
+                    reported_contest_result
+                ),
+                "auditedVotes": normalize_vals_for_discrepancies(
+                    audited_contest_result
+                ),
+                "discrepancies": normalize_vals_for_discrepancies(vote_deltas),
             }
 
     return discrepancies_by_jurisdiction
+
+
+def get_ballot_comparison_audit_discrepancies_by_jurisdiction(
+    election: Election, round_id: str
+) -> DiscrepanciesByJurisdiction:
+    discrepancies_by_jurisdiction: DiscrepanciesByJurisdiction = defaultdict(
+        lambda: defaultdict(dict)
+    )
+
+    ballots_in_round = (
+        SampledBallot.query.join(SampledBallotDraw)
+        .filter_by(round_id=round_id)
+        .distinct(SampledBallot.id)
+        .with_entities(SampledBallot.id)
+        .subquery()
+    )
+    sampled_ballot_id_to_jurisdiction_id = dict(
+        SampledBallot.query.filter(SampledBallot.id.in_(ballots_in_round))
+        .join(Batch)
+        .with_entities(SampledBallot.id, Batch.jurisdiction_id)
+    )
+    # make a readable identifier - it is the same default format as the ImprintedID
+    # but some sampled ballots have ImprintedIDs of a different format that are
+    # very long and unreadable, so here we ensure they all have the same format
+    # Ex. "Tabulator X, Batch Y, Ballot Position Z, Container 0"
```

Comment:
It might be more idiomatic if the frontend created this readable ID, but it makes the typing much more consistent with the batch comparison approach to create the ID here. Happy to get input on this design choice